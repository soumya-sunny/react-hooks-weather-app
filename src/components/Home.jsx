import React, { useState } from 'react';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import FormLabel from '@material-ui/core/FormLabel';
import Button from '@material-ui/core/Button';
import { useHistory } from "react-router-dom";


const useStyles = makeStyles((theme) => ({
    root: {
        position: "relative",
        marginTop: '65px',
        height:'400px'
    },
    selectFormControl:{
        width:'110px'
    }
}));


const Home = () => {
    const [city, setCity] = useState('');
    const history = useHistory();

    const getForecast =()=>{
        history.push("/forecast");
    }

    const handleChange = (event) => {
        setCity(event.target.value);
    };
    const classes = useStyles();
    return (
        <Grid container={true} className={classes.root} justify="space-around" spacing={2} alignItems="center" direction={"column"}>
            {/* <Grid container  sm={12} md={10} lg={10} justify="center"> */}
            <Typography variant="h6" noWrap>
                Choose your city
          </Typography>

            <FormControl className={classes.selectFormControl}>
                {/* <InputLabel id="demo-simple-select-label">Choose your city</InputLabel> */}
                <Select
                    labelId="demo-simple-select-label"
                    id="demo-simple-select"
                    value={city}
                    onChange={handleChange}
                >
                    <MenuItem value={10}>Bangalore</MenuItem>
                    <MenuItem value={20}>Gurgaon</MenuItem>
                    <MenuItem value={30}>London</MenuItem>
                    <MenuItem value={30}>Newyork</MenuItem>
                </Select>
            </FormControl>
            <FormControl component="fieldset">
                <RadioGroup row aria-label="unit" name="unit" defaultValue="celcius">
                    <FormControlLabel value="celcius" control={<Radio color="primary" />} label="Celcius" />
                    <FormControlLabel value="fahrenheit" control={<Radio color="primary" />} label="Fahrenheit" />
                </RadioGroup>
            </FormControl>
            <Button variant="contained" color="primary" onClick={getForecast}>
                Get Forecast
            </Button>
        </Grid>
    );
}

export default Home;