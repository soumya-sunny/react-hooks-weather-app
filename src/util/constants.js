export const forecastURL=`http://api.openweathermap.org/data/2.5/forecast?appid=${process.env.REACT_APP_API_KEY}&cnt=5&`;

//test