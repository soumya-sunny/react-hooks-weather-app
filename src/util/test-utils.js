import React from 'React';
import { Provider } from 'react-redux';
import { createStore } from "redux";
import rootReducer from "../redux/reducers";


const ReduxWrapper = ({children, state={}}) => {
    const store = createStore(rootReducer,state);
  
    return (
      <Provider store={store}> 
       {children}
      </Provider>
    )
  }
export {ReduxWrapper}