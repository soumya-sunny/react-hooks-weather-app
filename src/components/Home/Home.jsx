import React, { useState, useEffect } from 'react';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import FormLabel from '@material-ui/core/FormLabel';
import Button from '@material-ui/core/Button';
import { useHistory } from "react-router-dom";

import useFetch from "../../hooks/useFetch";
import { forecastURL } from "../../util/constants"


const useStyles = makeStyles((theme) => ({
    root: {
        position: "relative",
        marginTop: '65px',
        height:'400px'
    },
    selectFormControl:{
        width:'110px'
    }
}));


const Home = () => {
    const [city, setCity] = useState('');
    const [unit, setUnit] = useState('');
    const history = useHistory();

    const getForecastData = useFetch({
        // api: githubApi,
        method: "get",
        url: `${forecastURL}q=${city}&units=${unit}`,
    });

    const getForecast =()=>{
      getForecastData();
        history.push("/forecast");
    }

    const handleChange = (event) => {
        setCity(event.target.value);
    };
    const classes = useStyles();



        // api.openweathermap.org/data/2.5/forecast/daily?q=London&mode=xml&units=metric&cnt=7



    return (
        <Grid container={true} className={classes.root} justify="space-around" spacing={2} alignItems="center" direction={"column"}>
            <Typography variant="h6" noWrap>
                Choose your city
          </Typography>

            <FormControl className={classes.selectFormControl} variant='outlined'>
              
                <Select
                    labelId="demo-simple-select-label"
                    id="demo-simple-select"
                    value={city}
                    required
                    onChange={handleChange}
                    inputProps={{ "data-testid": "select-city" }}
                >
                    <MenuItem value={"Bengaluru"}>Bengaluru</MenuItem>
                    <MenuItem value={"Gurgaon"}>Gurgaon</MenuItem>
                    <MenuItem value={"London"}>London</MenuItem>
                    <MenuItem value={"New York"}>New York</MenuItem>
                </Select>
            </FormControl>
            <FormControl component="fieldset">
                <RadioGroup row aria-label="unit" name="unit" defaultValue="" onChange={e=>setUnit(e.target.value)}>
                    <FormControlLabel value="metric" control={<Radio color="primary" /> } label="Celcius" />
                    <FormControlLabel value="imperial" control={<Radio color="primary"/>} label="Fahrenheit" />
                </RadioGroup>
            </FormControl>
            <Button variant="contained" color="primary" onClick={getForecast} disabled={!city || !unit}>
                Get Forecast
            </Button>
        </Grid>
    );
}

export default Home;