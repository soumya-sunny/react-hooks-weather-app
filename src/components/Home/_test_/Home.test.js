import React from 'react'
import { render, fireEvent, screen } from '../../../util/test-utils'

import Home from '../Home'

describe('<Home />', () => {
    let wrapper;

    beforeEach(() => {
        const defaultProps = {}
        wrapper = render(<Home {...defaultProps} />)
    })

    test('render', () => {
        expect(wrapper).toMatchSnapshot()
    })

})
