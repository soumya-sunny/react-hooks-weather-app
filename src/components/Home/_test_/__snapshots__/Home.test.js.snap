// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
        id="grid"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
        >
          Choose your city
        </h6>
        <div
          class="MuiFormControl-root makeStyles-selectFormControl-2"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            required=""
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-label demo-simple-select"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              id="demo-simple-select"
              role="button"
              tabindex="0"
            >
              <span>
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              data-testid="select-city"
              required=""
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-4"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <fieldset
          class="MuiFormControl-root"
        >
          <div
            aria-label="unit"
            class="MuiFormGroup-root MuiFormGroup-row"
            role="radiogroup"
          >
            <label
              class="MuiFormControlLabel-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiRadio-root MuiRadio-colorPrimary MuiIconButton-colorPrimary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-10"
                    name="unit"
                    type="radio"
                    value="metric"
                  />
                  <div
                    class="PrivateRadioButtonIcon-root-11"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-12"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </div>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                Celcius
              </span>
            </label>
            <label
              class="MuiFormControlLabel-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiRadio-root MuiRadio-colorPrimary MuiIconButton-colorPrimary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-10"
                    name="unit"
                    type="radio"
                    value="imperial"
                  />
                  <div
                    class="PrivateRadioButtonIcon-root-11"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-12"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </div>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                Fahrenheit
              </span>
            </label>
          </div>
        </fieldset>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Get Forecast
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
      id="grid"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
      >
        Choose your city
      </h6>
      <div
        class="MuiFormControl-root makeStyles-selectFormControl-2"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          required=""
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="demo-simple-select-label demo-simple-select"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
            id="demo-simple-select"
            role="button"
            tabindex="0"
          >
            <span>
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            data-testid="select-city"
            required=""
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-4"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <fieldset
        class="MuiFormControl-root"
      >
        <div
          aria-label="unit"
          class="MuiFormGroup-root MuiFormGroup-row"
          role="radiogroup"
        >
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiRadio-root MuiRadio-colorPrimary MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-10"
                  name="unit"
                  type="radio"
                  value="metric"
                />
                <div
                  class="PrivateRadioButtonIcon-root-11"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-12"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Celcius
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiRadio-root MuiRadio-colorPrimary MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-10"
                  name="unit"
                  type="radio"
                  value="imperial"
                />
                <div
                  class="PrivateRadioButtonIcon-root-11"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-12"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Fahrenheit
            </span>
          </label>
        </div>
      </fieldset>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Get Forecast
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
